### PR TITLE
[WIP] Add prow deployment for acs-engine

### PR DESCRIPTION
**Pull request by 0xmichalis**

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
Why?

**Comment by kargakis** [comment]
We want to setup testing upstream.

File: cluster/ci/config/prow/acsengine/config.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+prowjob_namespace: acs-engine-ci
```
Comment (kargakis) [inline cluster/ci/config/prow/acsengine/config.yaml]: This needs to be tweaked to whatever namespace prow components are installed.

File: cluster/ci/config/prow/acsengine/config.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+tide:
+  target_url: https://deck-acs-engine-ci.svc.ci.openshift.org/tide.html
```
Comment (kargakis) [inline cluster/ci/config/prow/acsengine/config.yaml]: Deployment-specific, eg. http://acs-engine-prow.eastus.cloudapp.azure.com/tide.html

File: cluster/ci/config/prow/acsengine/config.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  queries:
+  - repos:
+    - kargakis/acs-engine
```
Comment (kargakis) [inline cluster/ci/config/prow/acsengine/config.yaml]: s/kargakis/Azure/

File: cluster/ci/config/prow/acsengine/ingress.yaml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+metadata:
+  name: tls-ing
+  annotations:
+    kubernetes.io/tls-acme: "true"
+    nginx.ingress.kubernetes.io/rewrite-target: /
+spec:
+  tls:
+  - secretName: prow-tls
+    hosts:
+    - acs-engine-prow.eastus.cloudapp.azure.com
```
Comment (kargakis) [inline cluster/ci/config/prow/acsengine/ingress.yaml]: Deployment-specific

File: cluster/ci/config/prow/acsengine/plugins.yaml [hunk 5]
```diff
@@ -0,0 +1,29 @@
+approve:
+- repos:
+  - kargakis/acs-engine
```
Comment (kargakis) [inline cluster/ci/config/prow/acsengine/plugins.yaml]: s/kargakis/Azure/

File: cluster/ci/config/prow/acsengine/plugins.yaml [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+config_updater:
+  maps:
+    cluster/ci/config/prow/acsengine/config.yaml:
+      name: config
+    cluster/ci/config/prow/acsengine/plugins.yaml:
+      name: plugins
+
+plugins:
+  kargakis/acs-engine:
```
Comment (kargakis) [inline cluster/ci/config/prow/acsengine/plugins.yaml]: s/kargakis/Azure/

File: cluster/ci/config/prow/acsengine/plugins.yaml [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+  - hold
+  - label
+  - lgtm
+  - lifecycle
+  - shrug
+  - size
+  - wip
+  - yuks
+
+  openshift/release:
```
Comment (kargakis) [inline cluster/ci/config/prow/acsengine/plugins.yaml]: Depending on where all these config files are stored, this needs to be updated accordingly.

Comment (CecileRobertMichon) [reply cluster/ci/config/prow/acsengine/plugins.yaml]: @kargakis does this mean I need to add the config files to the acs-engine repo?

Comment (kargakis) [reply cluster/ci/config/prow/acsengine/plugins.yaml]: You can store the config related to the deployment of prow (deployments, ingress, services) wherever you like.

The prow config file is more important and what most users are going to interface with. You should be able to store both config.yaml and plugins.yaml in acs-engine, so you likely want something like this:
```yaml
approve:
- repos:
  - Azure/acs-engine

config_updater:
  maps:
    .prowci/config.yaml:
      name: config
    .prowci/plugins.yaml:
      name: plugins

plugins:
  Azure/acs-engine:
  - approve
  - assign
  - cat
  - config-updater
  - dog
  - help
  - hold
  - label
  - lgtm
  - lifecycle
  - shrug
  - size
  - wip
  - yuks
```
We can walk through this and our own setup in our call next week.

**Pull request closed without merging**